### PR TITLE
Implement trigger

### PR DESCRIPTION
**Pull request by yongpingchen**
- Improve implementation of Command. Found problem of implementation of Command for aliasActionResults. The problem is when aliasActionResults array is null, after deserialized from Parcel, it become non null. 
- Make constructor of Trigger package private. Since SDK construct trigger using Gson, developer should not construct trigger by themselves. 
- Add small tests for Trigger.

### CODE REVIEW

File: thingif/src/test/java/com/kii/thingif/SmallTestBase.java [hunk 1]
```diff
@@ -47,4 +51,21 @@ protected void assertJSONArray(JSONArray expected, JSONArray actual) throws JSON
...
+    protected void assertSameCommands(Command expected, Command actual) {
+        Gson gson = new Gson();
+        Assert.assertEquals(gson.toJson(expected), gson.toJson(actual));
+
+        Assert.assertTrue(Arrays.equals(
+                expected.getAliasActions().toArray(),
+                actual.getAliasActions().toArray()));
+        Assert.assertFalse(
+                expected.getAliasActionResults() != null ^
+                actual.getAliasActionResults() != null );
```
Comment (shinsuke-nara) [inline thingif/src/test/java/com/kii/thingif/SmallTestBase.java]: This assertion is little bit difficult for me. How about following?

```java
Assert.assertTrue(
    expected.getAliasActionResults() != null && actual.getAliasActionResults() != null ||
    expected.getAliasActionResults() == null && actual.getAliasActionResults() == null)
```

Your code is very smart. My code is dull but I think it is more clear what we want to do.

**Comment by shinsuke-nara** [comment]
@yongpingchen Could you check my comment?

File: thingif/src/test/java/com/kii/thingif/SmallTestBase.java (hunk 1 above)
Comment (yongpingchen) [inline thingif/src/test/java/com/kii/thingif/SmallTestBase.java]: Sorry, it is better to use simple code.

**Comment by yongpingchen** [comment]
@shinsuke-nara please check the commit.

**Comment by shinsuke-nara** [comment]
@yongpingchen LGTM!

**Pull request merged**